qiskit_nature/algorithms/ground_state_solvers/minimum_eigensolver_factories/numpy_minimum_eigensolver_factory.py

```diff
@@ -45,30 +45,59 @@ def __init__(
         """
         self._filter_criterion = filter_criterion
         self._use_default_filter_criterion = use_default_filter_criterion
+        self._minimum_eigensolver = NumPyMinimumEigensolver(filter_criterion=filter_criterion)
 
     @property
     def filter_criterion(
         self,
     ) -> Callable[[Union[List, np.ndarray], float, Optional[List[float]]], bool]:
-        """returns filter criterion"""
+        """DEPRECATED. Use ``minimum_eigensolver`` method and solver properties instead.
+        Returns filter criterion."""
+        warnings.warn(
+            "The `filter_criterion` getter method is deprecated as of Qiskit Nature 0.4, "
+            "and will be removed in a future release. Use ``minimum_eigensolver`` method and "
+            "solver properties instead.",
+            DeprecationWarning,
+            stacklevel=2,
+        )
```

Comment:
I think you should be able to find a suitable method in `qiskit_nature.deprecation` for this purpose. I suggest `deprecate_property` which you can use as a decorator below the `@property` decorator. See here for an example: https://github.com/Qiskit/qiskit-nature/blob/fd016e2dc3774dedb41f934b37e389446435e27d/qiskit_nature/problems/second_quantization/base_problem.py#L62

EDIT: The same applies below